File: c++/DecoratorPattern/Tomato.cpp
#include "stdafx.h"
#include "Tomato.h"


Tomato::Tomato(Food *food) : Ingredients(food)
{
	this->Name.append("Î÷ºìÊÁ");
	this->Name.append(this->food->Name);
}


void Tomato::Cook()
{
	cout<<this->Name<<endl;
}

File: csharp/DesignPattern.Decorator/Tomato.cs

using System;
namespace DesignPattern.Decorator
{
    /// <summary>
    /// 西红柿
    /// </summary>
    public class Tomato : Ingredients
    {
        public Tomato(Food food)
            : base(food)
        {
            this.Name = "西红柿" + this.food.Name;
        }

        public override void Cook()
        {
            Console.WriteLine(this.Name);
        }
    }

}
